### PR TITLE
(MODULES-2207) Gem restrictions for Older puppet versions

### PR DESCRIPTION
**Pull request by ferventcoder**
Further restrict down the restrictions on windows dependencies when
using older Puppet versions, such that it works with Puppet 3.0.0 -
3.7.0. It does require a bit more logic in selecting gems but it allows
for testing older platforms without any additional known issues for
Puppet versions under 3.4.0.

### CODE REVIEW

File: .sync.yml [hunk 1]
```diff
@@ -18,6 +18,7 @@ Gemfile:
         version: '~>0.10.5'
     ':system_tests':
       - gem: beaker
+      - gem: beaker-rspec
```
Comment (cyberious) [inline .sync.yml]: This should not be here, we should put this in sync.yml when needed.

Comment (ferventcoder) [inline .sync.yml]: I'm 6 to one, half dozen to the other on this. Was more for supporting what @justinstoller might be calling in his scripts. Does it do harm if it is there?

**Comment by cyberious** [comment]
@ferventcoder can we clean up the commits of add and remove beaker-rspec

**Comment by ferventcoder** [comment]
Yes.

**Comment by ferventcoder** [comment]
Removed.

**Pull request merged**